platform/mbed_retarget.cpp

```diff
@@ -120,6 +121,17 @@ static void init_serial() {
 #endif
 }
 
+/**
+ * Sets errno when file opening fails.
+ * Wipes out the filehandle too.
+ */
+static int handle_open_errors(int error, unsigned filehandle_idx) {
+    errno = -error;
```

Comment:
why are we flipping the sign for errno? A comment might be nice as it is not clear